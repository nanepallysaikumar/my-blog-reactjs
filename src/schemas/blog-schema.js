import * as yup from 'yup';

export const blogSchema = yup.object().shape({
    header: yup
        .string()
        .min(3, 'Heading has to be longer than 3 characters!')
        .required('Heading is required!'),
    body: yup
        .string()
        .min(3, 'Body has to be longer than 3 characters!')
        .required('Body is required!')
});

export const loginSchema = yup.object().shape({
    email: yup
        .string()
        .email('Email is Invalid')
        .required('Email is required!'),
    password: yup.string().required('Password is required!')
});
