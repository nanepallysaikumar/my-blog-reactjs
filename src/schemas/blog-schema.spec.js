import { blogSchema } from './blog-schema';

describe('blog-schema', () => {
    it('should validate values', () => {
        const input = {
            header: 'test heading',
            body: 'test body'
        };
        return expect(blogSchema.validate(input)).resolves.toBe(input);
    });

    it('should check required properties', () => {
        const input = {};
        return expect(blogSchema.validate(input)).rejects.toMatchSnapshot(
            'blog-schema-required-error'
        );
    });

    it('should validate  minimum length of properties', () => {
        const input = {
            header: 'te',
            body: 'te'
        };
        return expect(blogSchema.validate(input)).rejects.toMatchSnapshot(
            'blog-schema-minimum-error'
        );
    });
});
