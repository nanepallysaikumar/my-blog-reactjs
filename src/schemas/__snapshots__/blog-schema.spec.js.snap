// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog-schema should check required properties: blog-schema-required-error 1`] = `[ValidationError: Body is required!]`;

exports[`blog-schema should validate  minimum length of properties: blog-schema-minimum-error 1`] = `[ValidationError: Body has to be longer than 3 characters!]`;
